core/state_connector_test.go

```diff
@@ -0,0 +1,268 @@
+package core
+
+import (
+	"errors"
+	"math/big"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/flare-foundation/coreth/core/types"
+	"github.com/flare-foundation/coreth/core/vm"
+	"github.com/flare-foundation/coreth/params"
+)
+
+type mockMessage struct {
+	from       common.Address
+	to         common.Address
+	gas        uint64
+	gasPrice   *big.Int
+	gasFeeCap  *big.Int
+	gasTipCap  *big.Int
+	value      *big.Int
+	data       []byte
+	accessList types.AccessList
+}
+
+func (m *mockMessage) From() common.Address         { return m.from }
+func (m *mockMessage) To() *common.Address          { return &m.to }
+func (m *mockMessage) GasPrice() *big.Int           { return m.gasPrice }
+func (m *mockMessage) GasFeeCap() *big.Int          { return m.gasFeeCap }
+func (m *mockMessage) GasTipCap() *big.Int          { return m.gasTipCap }
+func (m *mockMessage) Value() *big.Int              { return m.value }
+func (m *mockMessage) Gas() uint64                  { return m.gas }
+func (m *mockMessage) Nonce() uint64                { return 0 }
+func (m *mockMessage) Data() []byte                 { return m.data }
+func (m *mockMessage) AccessList() types.AccessList { return m.accessList }
+func (m *mockMessage) IsFake() bool                 { return true }
+
+type mockStateConnectorCaller struct {
+	context  vm.BlockContext
+	msg      *mockMessage
+	CallFunc func(context vm.BlockContext, caller vm.ContractRef, addr common.Address, input []byte, gas uint64, value *big.Int) (ret []byte, leftOverGas uint64, err error)
+}
+
+func (m *mockStateConnectorCaller) Call(caller vm.ContractRef, addr common.Address, input []byte, gas uint64,
+	value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+
+	return m.CallFunc(m.context, caller, addr, input, gas, value)
+}
+
+func (m *mockStateConnectorCaller) WithBlockContext(bc vm.BlockContext) { m.context = bc }
+
+func (m *mockStateConnectorCaller) BlockContext() vm.BlockContext { return m.context }
+
+func buildStateConnectorMock(from string, addr common.Address) (*mockMessage, *mockStateConnectorCaller) {
+	msg := mockMessage{
+		from: common.BytesToAddress([]byte(from)),
+	}
+	scc := mockStateConnectorCaller{
+		msg: &msg,
+		context: vm.BlockContext{
+			Coinbase: addr,
+		},
+	}
+	return &msg, &scc
+}
+
+func TestStateTransition_FinalisePreviousRound(t *testing.T) {
+	currentRoundNumber := []byte("222")
+	testCoinbaseAddress := common.BigToAddress(big.NewInt(1000))
+	blockTime := big.NewInt(time.Date(2022, time.February, 28, 17, 1, 1, 0, time.UTC).Unix())
+
+	t.Run("coston chain", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("nominal case", func(t *testing.T) {
+			t.Parallel()
```

Comment:
Not sure if this is legal 🤔 does it work as expected?

I usually only make the lowest level parallel, but if this works it's fine for me as is

Reply:
yes, It works :)